planner/core/cache.go

```diff
@@ -35,35 +34,18 @@ import (
 )
 
 var (
-	// preparedPlanCacheEnabledValue stores the global config "prepared-plan-cache-enabled".
-	// The value is false unless "prepared-plan-cache-enabled" is true in configuration.
-	preparedPlanCacheEnabledValue int32 = 0
-	// PreparedPlanCacheCapacity stores the global config "prepared-plan-cache-capacity".
-	PreparedPlanCacheCapacity uint = 1000
-	// PreparedPlanCacheMemoryGuardRatio stores the global config "prepared-plan-cache-memory-guard-ratio".
-	PreparedPlanCacheMemoryGuardRatio = 0.1
 	// PreparedPlanCacheMaxMemory stores the max memory size defined in the global config "performance-server-memory-quota".
 	PreparedPlanCacheMaxMemory = *atomic2.NewUint64(math.MaxUint64)
 )
 
 const (
-	preparedPlanCacheEnabled = 1
-	preparedPlanCacheUnable  = 0
+	preparedPlanCacheUnable uint32 = 0
 )
 
-// SetPreparedPlanCache sets isEnabled to true, then prepared plan cache is enabled.
-func SetPreparedPlanCache(isEnabled bool) {
-	if isEnabled {
-		atomic.StoreInt32(&preparedPlanCacheEnabledValue, preparedPlanCacheEnabled)
-	} else {
-		atomic.StoreInt32(&preparedPlanCacheEnabledValue, preparedPlanCacheUnable)
-	}
-}
-
 // PreparedPlanCacheEnabled returns whether the prepared plan cache is enabled.
 func PreparedPlanCacheEnabled() bool {
-	isEnabled := atomic.LoadInt32(&preparedPlanCacheEnabledValue)
-	return isEnabled == preparedPlanCacheEnabled
+	isEnabled := variable.PreparedPlanCacheSize.Load()
+	return isEnabled != preparedPlanCacheUnable
```

Comment:
How about removing `preparedPlanCacheUnable` as well and just using `variable.PreparedPlanCacheSize.Load() > 0` here?